relay-server/src/actors/store.rs

```diff
@@ -439,6 +446,54 @@ impl StoreForwarder {
         );
         Ok(())
     }
+
+    fn produce_replay_recording_chunks(
+        &self,
+        replay_id: EventId,
+        project_id: ProjectId,
+        item: &Item,
+    ) -> Result<ChunkedAttachment, StoreError> {
+        let id = Uuid::new_v4().to_string();
+
+        let mut chunk_index = 0;
+        let mut offset = 0;
+        let payload = item.payload();
+        let size = item.len();
+
+        // This skips chunks for empty replay recordings. The consumer does not require chunks for
+        // empty replay recordings. `chunks` will be `0` in this case.
+        while offset < size {
+            let max_chunk_size = self.config.attachment_chunk_size();
+            let chunk_size = std::cmp::min(max_chunk_size, size - offset);
+
+            let replay_recording_chunk_message =
+                KafkaMessage::ReplayRecordingChunk(ReplayRecordingChunkKafkaMessage {
+                    payload: payload.slice(offset, offset + chunk_size),
+                    replay_id,
+                    project_id,
+                    id: id.clone(),
+                    chunk_index,
+                });
+            self.produce(KafkaTopic::ReplayRecordings, replay_recording_chunk_message)?;
+            offset += chunk_size;
+            chunk_index += 1;
+        }
+
+        // The chunk_index is incremented after every loop iteration. After we exit the loop, it
+        // is one larger than the last chunk, so it is equal to the number of chunks.
+
+        Ok(ChunkedAttachment {
+            id,
+            name: REPLAY_RECORDINGS_ATTACHMENT_NAME.to_owned(),
```

Comment:
i don't forsee a use for the `name` field for us, but didn't seem worth defining our own type just to remove it.

Reply:
Keeping some constant name here is a good solution to distinguish the message contents in other places 👍 

If all you need here are `id`, `chunks`, and `size`, then you can also add those fields directly to your `ReplayRecordingKafkaMessage`, though.

Reply:
just went ahead and defined a separate type with just these 3 fields 👍🏼